### PR TITLE
skip intervals

### LINKED ISSUE #1499: Interval multiples
I like saying `interval: "year"` instead of `interval: d3.utcYear`. I’d like the same thing for `interval: d3.utcYear.every(10)`. So how about `interval: "5 years"` or more generally `N years` similar to what we do with the `p95` reducer?

### PR DESCRIPTION
**Pull request by mbostock**
Fixes #1499. Also allows plural interval names such as *years* instead of requiring strictly *year*. Uses d3.unixDay instead of d3.utcDay for the *day* interval so that the intervals are uniformly spaced, matching the current behavior of d3.utcTicks.

Adding some unit tests for maybeTimeInterval and maybeUtcInterval would be nice…

### CODE REVIEW

File: src/interval.d.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
+// For internal use.
+export type PeriodicTimeIntervalName =
+  | (`${number} ${TimeIntervalName}s` & Record<never, never>) // see https://github.com/microsoft/TypeScript/issues/29729
```
Comment (Fil) [inline src/interval.d.ts:3]:
```suggestion
  | ((`1 ${TimeIntervalName}` | `${number} ${TimeIntervalName}s`) & Record<never, never>) // see https://github.com/microsoft/TypeScript/issues/29729
```

"1 year" is accepted and that feels good, let's reflect it in the types?

File: test/plots/aapl-bollinger.ts [hunk 2]
```diff
@@ -41,9 +41,9 @@ export async function aaplBollingerGridSpacing() {
       Plot.gridY({interval: 10, stroke: "#fff", strokeOpacity: 1, strokeWidth: 0.5}),
       Plot.gridY({interval: 20, stroke: "#fff", strokeOpacity: 1}),
       Plot.axisY({interval: 20}),
-      Plot.gridX({interval: d3.utcMonth.every(3), stroke: "#fff", strokeOpacity: 1, strokeWidth: 0.5}),
-      Plot.gridX({interval: d3.utcYear, stroke: "#fff", strokeOpacity: 1}),
-      Plot.axisX({interval: d3.utcYear}),
+      Plot.gridX({interval: "3 months", stroke: "#fff", strokeOpacity: 1, strokeWidth: 0.5}),
+      Plot.gridX({interval: "1 years", stroke: "#fff", strokeOpacity: 1}),
+      Plot.axisX({interval: "1 years"}),
```
Comment (Fil) [inline test/plots/aapl-bollinger.ts:46]:
```suggestion
      Plot.gridX({interval: "1 year", stroke: "#fff", strokeOpacity: 1}),
      Plot.axisX({interval: "1 year"}),
```

File: src/interval.d.ts [hunk 3]
```diff
@@ -1,3 +1,9 @@
...
+  | "3 months"
+  | "10 years";
```
Comment (Fil) [inline src/interval.d.ts:5]:
```suggestion
  | "10 years"
  | "quarter" // alias for 3 months, no plural
  | "half" // alias for 6 months, no plural
;
```

File: src/interval.d.ts [hunk 4]
```diff
@@ -11,8 +17,8 @@ export type TimeIntervalName =
   | "day"
   | "week"
   | "month"
-  | "quarter"
-  | "half"
+  | "quarter" // alias for 3 months
+  | "half" // alias for 6 months
```
Comment (Fil) [inline src/interval.d.ts:21]:
```suggestion
```

File: src/time.js [hunk 5]
```diff
@@ -26,11 +24,9 @@ const utcIntervals = new Map([
   ["second", utcSecond],
   ["minute", utcMinute],
   ["hour", utcHour],
-  ["day", utcDay],
+  ["day", unixDay],
```
Comment (Fil) [inline src/time.js:27]: 👍

**Review by Fil — APPROVED** [review]
Suggestions to better reflect the possibilities ("1 year" is supported, "2 quarters" is not).

I've added a tiny bit of documentation, but we'll have to restructure it anyways together with the new quantize transform #1497.

**Comment by mbostock** [comment]
"2 quarters" is now supported.

**Pull request merged**